### ISSUE TITLE
Create a docker-ized version of OSM-NG

### ISSUE THREAD

**Issue by pomodoren** [openstreetmap-ng/openstreetmap-ng#98]
To simplify the process of development and deployment of osm-ng it makes sense to dockerize frontend, backend, and related services. A directory `/docker` that contains the `Dockerfile.backend` and others should be okay for this.

**Comment by Zaczero** [comment]
Yes, of course! Instead of a Dockerfile, it makes more sense to use a Nix builder, [like here](https://github.com/Zaczero/osm-relatify/blob/main/web/default.nix), to create a Docker image. This will result in zero maintenance, great reproducibility, and require no extra dependencies. Both the development environment and Docker image will then use the same source file to declare the available packages and services. Less room for error = less bugs 🙂.

**Comment by pomodoren** [comment]
Hey there! Great project!

To be honest, I think that for familiarity's sake, general low entry point for users who want to test the tool, and easy dev mode, docker and docker-compose can be a simpler choice. At least in the projects I am checking for transport digitalization, its standard the way how to get started: git clone X, cp .env.example .env, docker-compose up - and you are ready to go at localhost:xxxx.

Based on your shell.nix file I started a docker/ setup, will make a pull request - and you can see if it makes sense.

**Comment by Zaczero** [comment]
Adding `docker-compose.yml` is a good idea. Regarding Docker images, the plan is to automate Docker image updates using GitHub workflows. Then, Docker Compose could simply use those images. The application is not production-ready, so we haven't prioritized releases yet.
